src/main/kotlin/no/nav/tiltakspenger/saksbehandling/behandling/service/delautomatiskbehandling/DelautomatiskSoknadsbehandlingJobb.kt

```diff
@@ -0,0 +1,27 @@
+package no.nav.tiltakspenger.saksbehandling.behandling.service.delautomatiskbehandling
+
+import io.github.oshai.kotlinlogging.KotlinLogging
+import no.nav.tiltakspenger.libs.common.CorrelationId
+import no.nav.tiltakspenger.saksbehandling.behandling.ports.SøknadRepo
+import no.nav.tiltakspenger.saksbehandling.behandling.service.behandling.StartSøknadsbehandlingService
+
+class DelautomatiskSoknadsbehandlingJobb(
+    private val søknadRepo: SøknadRepo,
+    private val startSøknadsbehandlingService: StartSøknadsbehandlingService,
+    private val delautomatiskBehandlingService: DelautomatiskBehandlingService,
+) {
+    private val log = KotlinLogging.logger {}
+
+    suspend fun behandleNyeSoknader() {
+        val ubehandledeSoknader = søknadRepo.hentAlleUbehandledeSoknader(limit = 10)
+        log.info { "Fant ${ubehandledeSoknader.size} åpne søknader som det skal opprettes behandling for" }
+        ubehandledeSoknader.forEach {
```

Comment:
Ikke sikkert det er så viktig i første iterasjon, men hvis en av disse feiler uten å komme seg igjen, vil ingen søknader bli behandlet før en utvikler fikser det.

Blir litt samme som de rapids and rivers repoene som bare restarter appen og deadlocker på en poison pill

Reply:
Jeg har grublet litt frem og tilbake på dette og tror vi bør dele dette i to jobber slik at vi unngår å havne i en låst tilstand der steg en har gått bra og steg to har feilet.